src/libAtomVM/nifs.c

```diff
@@ -5842,6 +5868,118 @@ static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
     return result;
 }
 
+
+static term nif_lists_member(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+    term elem = argv[0];
+    term list = argv[1];
+    VALIDATE_VALUE(list, term_is_list);
+
+    int proper;
+    term_list_length(list, &proper);
+    if (UNLIKELY(!proper)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    while (!term_is_nil(list)) {
+        term head = term_get_list_head(list);
+
+        TermCompareResult cmp_result = term_compare(head, elem, TermCompareExact, ctx->global);
+
+        if (cmp_result == TermEquals) {
+            return TRUE_ATOM;
+        }
+
+        if (UNLIKELY(cmp_result == TermCompareMemoryAllocFail)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+
+        list = term_get_list_tail(list);
+    }
+
+    return FALSE_ATOM;
+}
+
+static term nif_lists_keymember(Context *ctx, int argc, term argv[])
+{
+    term result = nif_lists_keyfind(ctx, argc, argv);
+    return result == FALSE_ATOM ? FALSE_ATOM : TRUE_ATOM;
```

Comment:
`RAISE_ERROR` is defined as:
```c
 44 #define RAISE_ERROR(error_type_atom) \
 45     ctx->x[0] = ERROR_ATOM;          \
 46     ctx->x[1] = (error_type_atom);   \
 47     return term_invalid_term();
 ```
 
 So in case of error, `TRUE_ATOM` is returned. It is important to test all new NIFs also against invalid inputs, in order to check that we handle them properly and in a compliant way.